File: surface/widgets/compose/components/widgets/data-display/EmptyState.kt
// ============================================================
// Clef Surface Compose Widget — EmptyState
//
// Placeholder display shown when a data view contains no items.
// Provides a friendly visual with an optional icon, title,
// descriptive text, and call-to-action button. Compose
// adaptation: centered Column with icon, title Text, body Text,
// and Material 3 FilledTonalButton for the action.
// See widget spec: repertoire/widgets/data-display/empty-state.widget
// ============================================================

package clef.surface.compose.components.widgets.datadisplay

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.material3.FilledTonalButton
import androidx.compose.material3.Icon
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp

// --------------- Types ---------------

data class EmptyStateAction(
    val label: String,
    val onPress: () -> Unit,
)

// --------------- Component ---------------

/**
 * Placeholder display for empty data views with icon, message, and optional action.
 *
 * @param title Primary message explaining the empty state.
 * @param description Optional secondary text with guidance or context.
 * @param icon Optional icon displayed above the title.
 * @param action Optional call-to-action button.
 * @param modifier Modifier for the root layout.
 */
@Composable
fun EmptyState(
    title: String,
    description: String? = null,
    icon: ImageVector? = null,
    action: EmptyStateAction? = null,
    modifier: Modifier = Modifier,
) {
    Column(
        modifier = modifier
            .fillMaxWidth()
            .padding(horizontal = 32.dp, vertical = 48.dp),
        horizontalAlignment = Alignment.CenterHorizontally,
        verticalArrangement = Arrangement.Center,
    ) {
        // Icon
        if (icon != null) {
            Icon(
                imageVector = icon,
                contentDescription = null,
                modifier = Modifier.size(64.dp),
                tint = MaterialTheme.colorScheme.onSurfaceVariant,
            )
            Spacer(modifier = Modifier.height(16.dp))
        }

        // Title
        Text(
            text = title,
            style = MaterialTheme.typography.titleLarge,
            fontWeight = FontWeight.Bold,
            textAlign = TextAlign.Center,
        )

        // Description
        if (description != null) {
            Spacer(modifier = Modifier.height(8.dp))
            Text(
                text = description,
                style = MaterialTheme.typography.bodyMedium,
                color = MaterialTheme.colorScheme.onSurfaceVariant,
                textAlign = TextAlign.Center,
            )
        }

        // Action button
        if (action != null) {
            Spacer(modifier = Modifier.height(24.dp))
            FilledTonalButton(onClick = action.onPress) {
                Text(text = action.label)
            }
        }
    }
}

File: surface/widgets/nativescript/components/widgets/data-display/EmptyState.ts
// ============================================================
// Clef Surface NativeScript Widget — EmptyState
//
// Placeholder display for empty content areas.
// ============================================================

import { StackLayout, Label } from '@nativescript/core';
import type { View } from '@nativescript/core';

export interface EmptyStateProps {
  title?: string;
  description?: string;
  icon?: string;
  action?: View;
}

export function createEmptyState(props: EmptyStateProps): StackLayout {
  const { title = 'No data', description, icon, action } = props;
  const container = new StackLayout();
  container.className = 'clef-widget-empty-state';
  container.horizontalAlignment = 'center';
  container.padding = '24';

  if (icon) {
    const iconLabel = new Label();
    iconLabel.text = icon;
    iconLabel.fontSize = 48;
    iconLabel.horizontalAlignment = 'center';
    container.addChild(iconLabel);
  }

  const titleLabel = new Label();
  titleLabel.text = title;
  titleLabel.fontWeight = 'bold';
  titleLabel.fontSize = 18;
  titleLabel.horizontalAlignment = 'center';
  container.addChild(titleLabel);

  if (description) {
    const desc = new Label();
    desc.text = description;
    desc.textWrap = true;
    desc.horizontalAlignment = 'center';
    desc.opacity = 0.6;
    desc.marginTop = 8;
    container.addChild(desc);
  }
  if (action) { action.marginTop = 16; container.addChild(action); }
  return container;
}

export default createEmptyState;
